transmartApp/grails-app/controllers/org/transmartproject/interceptors/ConceptsInterceptor.groovy

```diff
@@ -1,12 +1,11 @@
 package org.transmartproject.interceptors
 
-import org.transmart.audit.AuditLogService
 import org.transmartproject.core.users.User
 
 
 class ConceptsInterceptor {
 
-    AuditLogService auditLogService
+    def auditLogService
```

Comment:
The AuditLogService is used to track events in an audit log. There are several interceptors that call into it, though now that I look at it they are all in transmartApp except for one usage in transmart-gwas-plugin. However one is the OauthInterceptor (that also needs to be moved I think).

Using a by type or by name bean is a dependency either way. The difference is that in one case it needs to be declared. Without further changes the interceptors that use this bean by-name will fail at runtime if it is not set.

The way the code is organised now this is a hard problem. Both transmartApp and transmart-oauth will need to use this service. I guess the architecturally correct solution is to create a new module for this, but I don't like the overhead of that for just a couple of classes.

Maybe the best solution is to create an AuditLogger interface for it in core-api and use that in the modules that use it with @Autowired(required=false), and make sure the code does not throw exceptions if it is not available. In that way the service can live in tm-server without a problem.